sound/soc/sof/intel/hda-dsp.c

```diff
@@ -371,8 +371,13 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 	if (target_state->substate == SOF_HDA_DSP_D0I3) {
 		value = SOF_HDA_VS_D0I3C_I3;
 
-		/* disable DMA trace in D0I3 */
-		flags = HDA_PM_NO_DMA_TRACE;
+		/*
+		 * Trace DMA can be kept enabled when the DSP enters
+		 * D0I3 while the system is in S0. It must be disabled
+		 * when the system is suspending.
```

Comment:
it will be disabled by the firmware when suspending to S0ix?

This part is still not clear, sorry.

Reply:
@plbossart the part that is not clear is about who disables the trace? Or are you asking to clarify how this is done by the FW?

Reply:
it's the "It must be disabled when the system is suspending' that is not clear.

if the device is already in S0/D0i3 and traces are on, and you suspend to S0ix/D0i3, then who would disable the trace?